server/src/main/java/io/deephaven/server/console/ConsoleServiceGrpcImpl.java

```diff
@@ -259,145 +261,203 @@ public void bindTableToVariable(BindTableToVariableRequest request,
-                            .setSuccess(true)
-                            .setRequestId(request.getRequestId())
-                            .addAllItems(results.stream().map(
-                                    // insertTextFormat is a default we used to set in constructor; for now, we'll just
-                                    // process the objects before sending back to client
-                                    item -> item.setInsertTextFormat(2).build())
-                                    .collect(Collectors.toSet()))
-                            .build();
-
-            safelyExecuteLocked(responseObserver,
-                    () -> responseObserver.onNext(AutoCompleteResponse.newBuilder()
-                            .setCompletionItems(mangledResults)
-                            .build()));
-        } catch (Exception exception) {
-            if (QUIET_AUTOCOMPLETE_ERRORS) {
-                if (log.isTraceEnabled()) {
-                    log.trace().append("Exception occurred during autocomplete").append(exception).endl();
-                }
-            } else {
-                log.error().append("Exception occurred during autocomplete").append(exception).endl();
+        @Override
+        public void onError(Throwable t) {
+            // ignore, client doesn't need us, will be cleaned up later
+        }
+
+        @Override
+        public void onCompleted() {
+            // just hang up too, browser will reconnect if interested
+            synchronized (responseObserver) {
+                responseObserver.onCompleted();
             }
-            safelyExecuteLocked(responseObserver,
-                    () -> responseObserver.onNext(AutoCompleteResponse.newBuilder()
-                            .setCompletionItems(GetCompletionItemsResponse.newBuilder()
-                                    .setSuccess(false)
-                                    .setRequestId(request.getRequestId()))
-                            .build()));
         }
     }
 
+    private static class NoopAutoCompleteObserver implements StreamObserver<AutoCompleteRequest> {
+        private final StreamObserver<AutoCompleteResponse> responseObserver;
+
+        public NoopAutoCompleteObserver(StreamObserver<AutoCompleteResponse> responseObserver) {
+            this.responseObserver = responseObserver;
+        }
+
+        @Override
+        public void onNext(AutoCompleteRequest value) {
+            // This implementation only responds to autocomplete requests with "success, nothing found"
+            if (value.getRequestCase() == AutoCompleteRequest.RequestCase.GET_COMPLETION_ITEMS) {
+                safelyExecuteLocked(responseObserver, () -> {
+                    responseObserver.onNext(AutoCompleteResponse.newBuilder()
+                            .setCompletionItems(
+                                    GetCompletionItemsResponse.newBuilder().setSuccess(true))
+                            .build());
```

Comment:
Need to set the `requestId` on the response.
```suggestion
                    responseObserver.onNext(AutoCompleteResponse.newBuilder()
                            .setCompletionItems(
                                    GetCompletionItemsResponse.newBuilder().setSuccess(true)
                                            .setRequestId(value.getGetCompletionItems().getRequestId()))
                            .build());
```